python/ray/dag/BUILD

```diff
@@ -115,7 +115,7 @@ py_test_module_list(
 )
 
 py_test_module_list(
-    size = "medium",
+    size = "large",
```

Comment:
I remembered that you mentioned the medium instance will make Redis tests fail. Do you know which part of this PR requires more resources?

Reply:
I think the tests just grew past the 300s threshold.

Reply:
Do you mean that increase the size here so that it can finish tests faster due to more resources?

Reply:
This should have the effect of increasing timeout for bazel.